lib/verify/response.rb

```diff
@@ -75,20 +75,21 @@ def address_missing?
     end
 
     def full_name
-      first_name = most_recent_verified_value(attributes.fetch("firstNames"))
-      middle_name = most_recent_verified_value(attributes.fetch("middleNames"))
-      surname = most_recent_verified_value(attributes.fetch("surnames"))
+      first_name = most_recent_verified_value(attributes.fetch("firstNames"), required: true)
+      middle_name = most_recent_verified_value(attributes.fetch("middleNames"), required: false)
+      surname = most_recent_verified_value(attributes.fetch("surnames"), required: true)
 
       [first_name, middle_name, surname].compact.join(" ")
     end
 
-    def most_recent_verified_value(attributes)
+    def most_recent_verified_value(attributes, required: true)
       return if attributes.blank?
 
-      attributes.sort_by { |attribute| attribute["from"].present? ? Date.strptime(attribute["from"], "%Y-%m-%d") : 0 }
+      verified_attribute = attributes.sort_by { |attribute| attribute["from"].present? ? Date.strptime(attribute["from"], "%Y-%m-%d") : 0 }
         .reverse
         .find { |attribute| attribute["verified"] }
-        .fetch("value")
+
+      required ? verified_attribute.fetch("value") : verified_attribute&.fetch("value")
```

Comment:
Perhaps we should now be more specific here with our error if we get back a Verify response that has a required field missing? for example:

```ruby

class MissingResponseAttribute < StandardError; end
...
raise MissingResponseAttribute, "No verified value found for #{attributes}" if required && verified_attribute.nil?
verified_attribute.fetch("value")
```